docs/dev-guide/3.0-development/index.rst

```diff
@@ -0,0 +1,17 @@
+.. warning::
+    All documents within the 3.0-Development section should be considered temporary.
```

Comment:
I think we only need to have this adminition in the 3.0-dev index file. The other instances of it can go away.

Also, defining "temporary" might be nice, e.g. 'All documents within the 3.0 Development section should be considered temporary, and will be removed or relocation prior to the release of Pulp 3.0.'